src/config.rs

```diff
@@ -0,0 +1,50 @@
+use std::time::Duration;
+
+use clap::{Parser, Subcommand};
+
+/// Available DNS providers as subcommands, each with its own configuration options
+#[derive(Clone, Debug, Subcommand)]
+pub enum DnsProvider {
+    /// Hetzner DNS provider configuration
+    Hetzner(HetznerConfig),
+}
+
+/// Define a struct to hold all command-line arguments
+#[derive(Clone, Debug, Parser)]
+#[command(author, about, version)]
+pub struct Config {
+    /// Specifies the address of the Consul server.
+    #[arg(long, default_value = "localhost:8500")]
+    pub consul_address: url::Url,
+
+    /// Optionally sets the datacenter of the Consul server.
+    #[arg(long)]
+    pub consul_datacenter: Option<String>,
+
+    /// Consul TTL times out after this duration without being renewed
+    #[arg(long, default_value = "30s", value_parser = humantime::parse_duration)]
+    pub consul_ttl: Duration,
```

Comment:
I don't think it makes sense to expose this to the user as it controls something very internal to how the application works.

Reply:
removed